Packs/SymantecEndpointSecurity/Integrations/SymantecEndpointSecurity/SymantecEndpointSecurity.py

```diff
@@ -0,0 +1,459 @@
+            )
+            raise UnauthorizedToken
+        elif e.res is not None and e.res.status_code == 410:
+            raise NextPointingNotAvailable
+        raise e
+
+    for chunk in json_res:
+        events.extend(chunk["events"])
+    next_hash = json_res[0].get("next", "") if json_res else ""
+
+    if not events:
+        demisto.info("Not events returned")
+        return
+
+    demisto.info(f"Starting event filtering. Initial number of events: {len(events)}")
+    filtered_events = filter_duplicate_events(events, integration_context)
+    demisto.info(f"Filtering completed. Total number of events: {len(filtered_events)}")
+
+    filtered_events.extend(integration_context.get("last_fetch_events", []))
+
+    try:
+        push_events(filtered_events)
+        demisto.info("pushing passed successfully")
+    except Exception as e:
+        update_new_integration_context(
+            filtered_events=filtered_events,
+            next_hash=next_hash,
+            include_last_fetch_events=True,
+            last_integration_context=integration_context,
+        )
+        demisto.info(
+            f"Failed to push events to XSIAM, The integration_context updated. Error: {e}"
+        )
+        raise e
+
+    update_new_integration_context(
+        filtered_events=filtered_events,
+        next_hash=next_hash,
+        include_last_fetch_events=False,
+        last_integration_context=integration_context,
+    )
+
+
+def perform_long_running_loop(client: Client):
+    while True:
+        # Used to calculate the duration of the fetch run.
+        start_run = get_current_time_in_seconds()
+        try:
+            integration_context = get_integration_context()
+            demisto.info(f"Starting new fetch with {integration_context=}")
+            get_events_command(client, integration_context=integration_context)
+
+        except UnauthorizedToken:
+            try:
+                # Used to calculate the duration of the fetch run.
+                end_run = get_current_time_in_seconds()
+                manage_fetch_interval(client, start_run, end_run)
+                # Trying to obtain a new access token
+                client.get_token()
+            except Exception as e:
+                demisto.info("Failed to obtain a new access token")
+                raise e
+            continue
+        except NextPointingNotAvailable:
+            demisto.info("The next hash not available, pop it from integration context")
+            integration_context.pop("next_fetch")
+            set_integration_context(integration_context)
+
+            # Used to calculate the duration of the fetch run.
+            end_run = get_current_time_in_seconds()
+            manage_fetch_interval(client, start_run, end_run)
+            continue
+        except Exception as e:
+            demisto.info(f"Failed to fetch logs from API. Error: {e}")
+            raise e
```

Comment:
`raise DemistoException("here be message") from e` instead of logging and reraising